backend/src/middleware/notifications/notificationsMiddleware.spec.js

```diff
@@ -428,7 +458,7 @@ describe('notifications', () => {
           beforeEach(async () => {
             commentContent =
               'One mention about me with <a data-mention-id="you" class="mention" href="/profile/you" target="_blank">@al-capone</a>.'
```

Comment:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2020-02-01T14:23:50Z" title="Saturday, February 1st 2020, 3:23:50 pm +01:00">Feb 1, 2020</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -541,5 +573,148 @@ describe('notifications', () => {
         })
       })
     })
+
+    describe('given I file a report on a', () => {
+      let resourceId
+      const reasonCategory = 'discrimination_etc'
+      const reasonDescription = 'I am free to be gay !!!'
+      let reportFiler
+      let reportedUserOrAuthorData
+      const fileReportAction = async () => {
+        authenticatedUser = await reportFiler.toJson()
+        await mutate({
+          mutation: fileReportMutation,
+          variables: {
+            resourceId,
+            reasonCategory,
+            reasonDescription,
+          },
+        })
+        authenticatedUser = await notifiedUser.toJson()
+      }
+      const setExpectedNotificationOfReportedResource = reportedResource => {
+        return expect.objectContaining({
+          data: {
+            notifications: [
+              {
+                createdAt: expect.any(String),
+                read: false,
+                reason: 'filed_report_on_resource',
+                from: {
+                  __typename: 'Report',
+                  id: expect.any(String),
+                  filed: [
+                    {
+                      reasonCategory: 'discrimination_etc',
+                      reasonDescription: 'I am free to be gay !!!',
+                      reportedResource,
+                    },
+                  ],
+                },
+              },
+            ],
+          },
+        })
+      }
+
+      beforeEach(async () => {
+        reportFiler = notifiedUser
+        reportedUserOrAuthorData = {
+          id: 'reportedUser',
+          name: 'Mrs Badman',
+          slug: 'mrs-badman',
+          email: 'reported-user@example.org',
+          password: '1234',
+        }
+      })
+
+      describe('user', () => {
+        it('sends me a notification for filing a report on a user', async () => {
+          await neode.create('User', reportedUserOrAuthorData)
```


Better use our factories here, e.g. `email` will be stored in the node, but we use a separate node for a long time already.

Reply:
> [<img alt="Tirokk" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2020-02-06T07:01:18Z" title="Thursday, February 6th 2020, 8:01:18 am +01:00">Feb 6, 2020</time>_

---


So true. @roschaefer 
I refactored all the tests and removed neode completely from this test.